### ISSUE TITLE
@TimeSeries collection is ceated as a "collection" type on startup

### ISSUE THREAD

**Issue by salvatorenovelli** [spring-projects/spring-data-mongodb#4792]
Hi there, 
according to [Index and Collection Management document page](https://docs.spring.io/spring-data/mongodb/reference/mongodb/template-collection-management.html), timeseries collection should be created before inserting data to make sure they're created as a `type` = "timeseries". (instead of a plain "collection")

If no annotation is present on the class, creating the collection using `MongoTemplate` and `CreateCollectionOptions` works well, but when I add the `@Timeseries` annotation and I try to create it using this is the code, from Spring Data Mongo documentation page, I get an error, saying the collection already exist:
  
    @TimeSeries(collection="weather", timeField = "timestamp")
    public class Measurement {

    String id;
    Instant timestamp;
    // ...
    }

    template.createCollection(Measurement.class);

This seems to be related to the fact that collections are automatically created by `MongoPersistentEntityIndexCreator` to ensure indexes are created, but this creator seems to ignore the fact the the collection is a TimeSeries and creates it as a plain "collection".

The work-around is quite simple, removing the annotation gives you a chance to create it manually before any insert but it doesn't feel the most consistent way given all my other collections are correctly created and configured via annotations.

**Comment by christophstrobl** [comment]
Thank our for the report. Sounds like we should update the documentation stressing that enabling auto index creation (disabled by default) can interfere with manual collection creation.

**Comment by salvatorenovelli** [comment]
Fair enough as a position but auto-index creation is a very neat feature. It's not just about creating indexes in a new database, it's also about failing fast, with meaningful exceptions if indexes are created in the wrong way in an existing database. Auto-index creation allows for a reliable database configuration but also run-time assertions based on annotations specified on a single place.

Another option you could consider, if valuable to enough users, could be to provide an option to disable auto-index creation specifically for time-series collections.
